src/special-types-and-traits.md

```diff
@@ -110,7 +110,7 @@ according to the following rules:
   closure that captures a `T` by shared reference and a `U` by value implements
   any auto traits that both `&T` and `U` do.
 
-For generic types (counting the built-in types above as generic over `T`), if an
+For generic types (counting the built-in types above as generic over `T`), if a
 generic implementation is available, then the compiler does not automatically
 implement it for types that could use the implementation except that they do not
```

Comment:
@Centril Perhaps this can be re-phrased? I can't seem to understand what the `except that they do not meet...` part means. Does that mean `except if they` or `except then they`...

Reply:
Consider:
```rust
struct A<T>(T);

unsafe impl<T: Copy> Send for A<T> {}

fn assert_send<T: Send>() {}

fn main() {
    assert_send::<A<u8>>();
    assert_send::<A<String>>();
}
```

Reply:
So the compiler would have done this  (and does so if we comment out our `unsafe impl`):
```rust
unsafe impl Send for A<String> {}
// as part of:
unsafe impl Send for A<T> where T: Send {}
```

But our `unsafe impl` tells the compiler to disregard its `auto trait` rules and instead use our "rule" (`impl`), i.e. we tell the compiler that it is not safe to `Send` values of type `A<T>` unless `T: Copy`?

Reply:
Yep, pretty much.